pkg/queue/queue.go

```diff
@@ -109,19 +115,40 @@ func (cq *ClusterQueue) PushIfNotPresent(info *workload.Info) bool {
 	if item != nil {
 		return false
 	}
+
+	// Populate priority from priority class before enqueue.
+	p, err := workload.GetPriorityFromPriorityClass(context.Background(),
+		cq.client, info.Obj.Spec.PriorityClassName)
+	// Since the client is cached, the only possible error is NotFound
+	if apierrors.IsNotFound(err) {
+		return false
+	}
+	info.Priority = p
+
 	heap.Push(&cq.heap, *info)
 	return true
 }
 
-func (cq *ClusterQueue) PushOrUpdate(w *kueue.QueuedWorkload) {
+func (cq *ClusterQueue) PushOrUpdate(w *kueue.QueuedWorkload) bool {
 	item := cq.heap.items[workload.Key(w)]
 	info := *workload.NewInfo(w)
+
+	// Populate priority from priority class before enqueue.
+	p, err := workload.GetPriorityFromPriorityClass(context.Background(),
+		cq.client, info.Obj.Spec.PriorityClassName)
+	// Since the client is cached, the only possible error is NotFound
+	if apierrors.IsNotFound(err) {
+		return false
+	}
```

Comment:
lets encapulate this in a function that return the default priority when not found and logs an error; and use the function above as well.